raft-tech/TANF-app
# Feature/2729 remote migrations

## Summary of Changes
Pull request closes #2729 

* Enables migrations to be run remotely from a circleci server, rather than coupled with the app startup in cloud.gov

## How to Test
1. Create a migration locally that doesn't exist on the server. Commit and push the change
    * a failing migration can be created by copying any other migration, renaming it to the next number in the sequence (i.e., `0039_migration_name.py` becomes `0040_new_migration_name.py`) and making no other changes. This should produce a conflict as two migrations share the same dependency 
3. Use the `tdrs-deploy` helper command to deploy the branch
4. In circleci, observe the "Apply remote migrations" step
    * if the migration was supposed to be successful, ensure it passes and the migration is applied to the server
    * if the migration was supposed to fail, ensure the ci task exits with an error, the migration is not applied, and the deployment does not take place

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
+ [x] New migrations apply ahead of deploying app
    - Confirm by creating a migration of a large dataset

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
